grails-app/controllers/org/broadinstitute/orsp/CommentsController.groovy

```diff
@@ -1,13 +1,57 @@
 package org.broadinstitute.orsp
 
+import grails.converters.JSON
+import grails.rest.Resource
+import groovy.util.logging.Slf4j
+
+@Slf4j
+@Resource(readOnly = false, formats = ['JSON', 'APPLICATION-MULTIPART'])
 class CommentsController extends AuthenticatedController {
 
+    CommentsService commentsService
+
     def index = { list }
 
+    @Deprecated
     def list = {
         Issue issue = queryService.findByKey(params.id)
         Collection<Comment> comments = Comment.findAllByProjectKey(params.id)
         [issue: issue, comments: comments]
     }
 
+    def saveNewComment() {
+        try {
+            Comment savedComment = commentsService.addComment(params.id, params.comment)
+            response.status = 200
+            render savedComment as JSON
+        } catch (IllegalArgumentException e) {
+            response.status = 400
+            render([error: e.message] as JSON)
+        } catch (Exception e) {
+            response.status = 500
+            render([error: e.message] as JSON)
+        }
+    }
+
+    def getComments() {
+        JSON.registerObjectMarshaller(Comment) {
+            def output = [:]
+            output['id'] = it.id
+            output['author'] = it.author
+            output['comment'] = it.description
+            output['date'] = it.created
+            return output
+        }
+        try {
+            Collection<Comment> comments = commentsService.getCommentsForIssueId(params.id)
+            response.status = 200
+            render comments as JSON
+        } catch(IllegalArgumentException e) {
+            response.status = 400
+            render([error: e.message] as JSON)
+        } catch(Error e) {
+            response.status = 500
+            render([error: e.message] as JSON)
```

Comment:
Not for this PR ... definitely a tech-debt ticket. We should abstract these error conditions even more than we're doing now. I think it would be nice to let the super-class handle the different common exception cases [the way we do in consent](https://github.com/DataBiosphere/consent/blob/develop/src/main/java/org/broadinstitute/consent/http/resources/Resource.java#L59). The object we render should also be an `Error` class that has a status and whatever exception is thrown.

Reply:
Indeed this would be an improvement. I created a tech debt story for this.
[BTRX-737](https://broadinstitute.atlassian.net/browse/BTRX-737)